pkg/reconciler/instances/istio/reset/pod/handler.go

```diff
@@ -112,10 +134,162 @@ func (i *RolloutHandler) ExecuteAndWaitFor(context context.Context, object Custo
+		}
 		return err
 	}
 
-	return i.WaitForResources(context, object)
+	return nil
+}
+
+func annotateObject(context context.Context, client kubernetes.Interface, object CustomObject, key string, val string) error {
+	switch object.Kind {
+	case "Deployment":
+		err := annotatePodsFor(context, getDeploymentLabelSelector, client, object, key, val)
+		if err != nil {
+			return err
+		}
+	case "DaemonSet":
+		err := annotatePodsFor(context, getDaemonSetLabelSelector, client, object, key, val)
+		if err != nil {
+			return err
+		}
+	case "ReplicaSet":
+		err := annotatePodsFor(context, getReplicaSetLabelSelector, client, object, key, val)
+		if err != nil {
+			return err
+		}
+
+	case "StatefulSet":
+		err := annotatePodsFor(context, getStatefulSetLabelSelector, client, object, key, val)
+		if err != nil {
+			return err
+		}
+	case "Job":
+		err := annotatePodsFor(context, getJobLabelSelector, client, object, key, val)
+		if err != nil {
+			return err
+		}
+	default:
+		err := fmt.Errorf("kind %s not found", object.Kind)
+		return err
+	}
+	return nil
+}
+
+type labelSelectorGetter func(context context.Context, client kubernetes.Interface, object CustomObject) (string, error)
+
+func annotatePodsFor(context context.Context, f labelSelectorGetter, client kubernetes.Interface, object CustomObject, annotationKey, annotationVal string) error {
+	labelSelector, err := f(context, client, object)
+	if err != nil {
+		return err
+	}
+	podList, err := listPodsWithLabelSelector(context, client, object.Namespace, labelSelector)
+	if err != nil {
+		return err
+	}
+	err = annotatePods(context, client, podList, annotationKey, annotationVal)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func listPodsWithLabelSelector(context context.Context, client kubernetes.Interface, namespace string, labelSelector string) (*v1.PodList, error) {
+	podList, err := client.CoreV1().Pods(namespace).List(context, metav1.ListOptions{LabelSelector: labelSelector})
+	if err != nil {
+		return nil, err
+	}
+	return podList, nil
+}
+
+func annotatePods(context context.Context, kubeClient kubernetes.Interface, podList *v1.PodList, key string, timeout string) error {
+	for _, pod := range podList.Items {
+		if pod.Annotations == nil {
+			pod.Annotations = make(map[string]string)
+		}
+		pod.Annotations[key] = timeout
+		_, err := kubeClient.CoreV1().Pods(pod.Namespace).Update(context, &pod, metav1.UpdateOptions{}) //nolint:gosec
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func getJobLabelSelector(context context.Context, client kubernetes.Interface, object CustomObject) (string, error) {
+	job, err := client.BatchV1().Jobs(object.Namespace).Get(context, object.Name, metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+	labelSelector := buildLabelSelector(job.Spec.Selector.MatchLabels)
+	return labelSelector, nil
+}
+
+func getDeploymentLabelSelector(context context.Context, client kubernetes.Interface, object CustomObject) (string, error) {
```

Comment:
I have a feeling that this name is kind of misleading. When one uses `get` verb then we should safely assume that there is some underlying property returned or max one external call made. In fact what you do is you fetch labels from the cluster and perform some custom logic on them. What about calling it `build `the way that you did for the inner private method?